src/typelib.erl

```diff
@@ -4,6 +4,7 @@
 -export([remove_pos/1, annotate_user_types/2, get_module_from_annotation/1,
          substitute_type_vars/2,
          pp_type/1, debug_type/3, parse_type/1]).
+-export_type([constraint/0, function_type/0, type_et_cetera/0]).
```

Comment:
The name "type_et_cetera" is not very good. I added it just very temporarily and locally. When exporting it, I think it would make sense to find a better name. How about "extended_type"? Is that better?

This type itself isn't very good IMO. Especially, I don't like that intersection is represented simply as a list. We should change the representation of intersection to `{type, _, intersection [our_type()]}` though the code, but that's another task.